lib/stdlib/src/epp.erl

```diff
@@ -978,14 +983,29 @@ scan_toks(Toks0, From, St) ->
 	    wait_req_scan(St)
     end.
 
+%% Reads the content of the file and rewrites the AST as if
+%% the content had been written in-place.
+scan_filedoc([{'{',_}, {atom, _,file},
+                {',', _}, {string, A, DocFilename},
+                {'}', _}, {dot, _}=End], [_, ModuleDoc]=Module, From, #epp{name = CurrentFilename}=St) ->
+    Filename = filename:join(filename:dirname(CurrentFilename), DocFilename),
+    case file:read_file(Filename) of
+        {ok, Bin} ->
+            epp_reply(From, {ok, Module ++ [{string, A, binary_to_list(Bin)}, End]}),
```

Comment:
The "anno" for this string should be the file that was read, not the original anno. Maybe you also need to emit some `{file,...}` attributes?

This so that in the end the anno of the doc in the docchunk becomes `path/to/file.md` instead of `path/to/src.erl`. ExDoc uses this information to generate warnings for the docs and provide links for editing the docs.

Reply:
also, it should be possible to write `-doc({file, "..."})`. The parenthesis are optional.